modules/host/storageobligations.go

```diff
@@ -732,7 +757,9 @@ func (h *Host) handleActionItem(so *storageObligation) {
 		sectorIndex := segmentIndex / (modules.SectorSize / crypto.SegmentSize)
 		// Pull the corresponding sector into memory.
 		sectorRoot := so.SectorRoots[sectorIndex]
+		h.mu.Lock()
 		sectorBytes, err := h.ReadSector(sectorRoot)
+		h.mu.Unlock()
```

Comment:
I think I was just being over-eager with the locking. I don't think we even need a readlock here.